Src/IronPython/Runtime/ByteArray.cs

```diff
@@ -364,23 +364,60 @@ public ByteArray center(int width, [BytesLike, NotNull]IList<byte> fillchar)
             }
         }
 
-        public int find([BytesLike, NotNull]IList<byte> sub) => find(sub, null, null);
+        public int find([BytesLike, NotNull]IList<byte> sub) {
+            lock (this) {
+                return _bytes.Find(sub, 0, _bytes.Count);
+            }
+        }
 
-        public int find([BytesLike, NotNull]IList<byte> sub, int start) => find(sub, start, null);
+        public int find([BytesLike, NotNull]IList<byte> sub, int start) {
+            lock (this) {
+                return _bytes.Find(sub, start, _bytes.Count);
+            }
+        }
 
-        public int find([BytesLike, NotNull]IList<byte> sub, int? start, int? end) {
+        public int find([BytesLike, NotNull]IList<byte> sub, int start, int end) {
             lock (this) {
                 return _bytes.Find(sub, start, end);
             }
         }
 
-        public int find(int @byte) => find(@byte, null, null);
+        public int find([BytesLike, NotNull]IList<byte> sub, object? start)
+            => find(sub, start, null);
+
+        public int find([BytesLike, NotNull]IList<byte> sub, object? start, object? end) {
+            int istart = start != null ? Converter.ConvertToIndex(start) : 0;
+            lock (this) {
+                int iend = end != null ? Converter.ConvertToIndex(end) : _bytes.Count;
+                return _bytes.Find(sub, istart, iend);
+            }
+        }
+
+        public int find(BigInteger @byte) {
+            lock (this) {
+                return _bytes.IndexOfByte(@byte.ToByteChecked(), 0, _bytes.Count);
```

Comment:
Unrelated to this PR, but we should update `IListOfByteOps.IndexOfByte` to have a `byte` argument.